xds/internal/xdsclient/xdsresource/tests/unmarshal_cds_test.go

```diff
@@ -103,6 +104,8 @@ func (s) TestValidateCluster_Success(t *testing.T) {
 	defer func() {
 		envconfig.XDSCustomLBPolicy = origCustomLBSupport
 	}()
+	defer func(old bool) { envconfig.LeastRequestLB = old }(envconfig.LeastRequestLB)
+	envconfig.LeastRequestLB = true
```

Comment:
Do you need a test to ensure that when the env var is not set, we dont look at the config?

Reply:
As crazy as this is, yes: https://github.com/grpc/grpc-go/blob/e5f50f251f9824c7f2e0b3e6dc8b813af6664c8a/xds/internal/xdsclient/xdsresource/unmarshal_cds_test.go#L208. I updated name though to be more relevant/state the updated root cause :).